twilight-validate/src/component.rs

```diff
@@ -501,9 +565,38 @@ pub enum ComponentValidationErrorType {
         /// Provided number of codepoints.
         chars: usize,
     },
+    /// V2 Components used in a V1 component.
+    DisallowedV2,
```

Comment:
I think

```suggestion
    DisallowedV2Components,
```

is probably more future-proof and self-explanatory